radix-engine/src/system/system_db_reader.rs

```diff
@@ -140,6 +285,254 @@ impl<'a, S: SubstateDatabase> SystemDatabaseReader<'a, S> {
         definition.value.map(|v| v.into_latest())
     }
 
+    pub fn get_kv_store_type_target(&self, node_id: &NodeId) -> Option<KVStoreTypeTarget> {
+        let type_info = self.fetch_substate::<SpreadPrefixKeyMapper, TypeInfoSubstate>(
+            node_id,
+            TYPE_INFO_FIELD_PARTITION,
+            &TypeInfoField::TypeInfo.into(),
+        )?;
+
+        let kv_store_info = match type_info {
+            TypeInfoSubstate::KeyValueStore(kv_store_info) => kv_store_info,
+            _ => return None,
```

Comment:
No invariant is broken here (except for if the client passes in an erroneous node_id) so would rather keep this. More long term, this should be a `Result::Err` of some kind

Reply:
I think it's more about the contract of the interface.

The method says `get_kv_store_type_target` so the callers expect a kv store, so it can't tell if it's missing OR it's of wrong type.

This is similar to `get_blueprint_id` which will panic if it's not an object (of blueprint).

Reply:
I've updated interface to return more specific errors now